toolkit/tools/scheduler/schedulerutils/buildworker.go

```diff
@@ -193,11 +196,41 @@ func buildSRPMFile(agent buildagents.BuildAgent, buildAttempts int, srpmFile, ou
 	)
 
 	logBaseName := filepath.Base(srpmFile) + ".log"
-	err = retry.Run(func() (buildErr error) {
-		builtFiles, logFile, buildErr = agent.BuildPackage(srpmFile, logBaseName, outArch, dependencies)
-		return
-	}, buildAttempts, retryDuration)
-
+	const maxAttempts := 3
+	numAttempts := buildAttempts
```

Comment:
Instead of hard-coding values, I feel like we should differentiate between regular build failures, for which `buildAttempts` is used, and successful builds with failing tests. We shouldn't be silently overriding the rebuild count passed by the user.

My suggestions:
- Add a new parameter controlling the number of retries we want to perform in case of test failures. We could then first apply the build retries in case of a build failure and if the build succeeds, then we'd look into test results and their retry counts.
- [**My preference**] Remove retrying failed package builds and use only test retries. I think we never actually used anything than a single build re-try (the package build have been pretty stable) and in general, I believe we shouldn't be hiding flaky package builds and instead we should be fixing them.

Reply:
Essentially, the tests are failing due to timing out, but I haven't found any way to *only* run the %check section of a spec file, which has made it hard to only run tests again without building. I guess I could theoretically use the `rpmbuild` flags `-bi --short-circuit`, but that would still be re-installing before running the tests again.

To make sure I'm understanding what you're saying, are you saying that I should leave the `retry.Run` as is and let the package build run, and then parse the tests after, and rebuild if and only if the tests fail? Or do you mean that I should change some of my if-case logic to only check for test failures? Either way, I don't see a way to test without rebuilding/reinstalling the built RPM.

Reply:
Yes, I was thinking about the former, so re-running the full build in case of a test failure (and still failing immediately after first build failure). You're right that there's no easy way to retry only the `%check` section. Or at least no easy way I can think of right now. Anyway, since test builds are not time-critical, we might be fine with full rebuilds for now. If we end up timing out the whole build, then we'll need to improve.

Reply:
Okay, that makes sense. I was able to get retry.Run working correctly now, which makes me think something might have been wrong with `redis` and not `retry.Run`, so that's being handled in its own PR.